packages/webapp/src/components/Filter/FilterMultiSelectV2/index.tsx

```diff
@@ -0,0 +1,251 @@
+import TextButton from '../../Form/Button/TextButton';
+import styles from './styles.module.scss';
+import Checkbox from '../../Form/Checkbox';
+import produce from 'immer';
+import { FilterItemProps } from '../FilterGroup';
+import { Label } from '../../Typography';
+
+interface FilterMultiSelectProps extends ComponentFilter {
+  onChange?: FilterItemProps['onChange'];
+  isDisabled?: boolean;
+}
+
+const MultiValueRemove = (props: MultiValueRemoveProps) => (
+  <components.MultiValueRemove {...props}>
+    <XCircle />
+  </components.MultiValueRemove>
+);
+
+const ClearIndicator = (props: ClearIndicatorProps<ComponentFilterOption>) => {
+  const { getValue, clearValue } = props;
+  const value = getValue();
+  const { t } = useTranslation();
+
+  const onClick = (e: MouseEvent) => {
+    e.stopPropagation();
+    e.preventDefault();
+    clearValue();
+  };
+
+  return (
+    <>
+      {t('FILTER.SELECTED', { count: value.length })} -
+      <TextButton className={styles.clearTextButton} onClick={onClick}>
+        clear
+      </TextButton>
+    </>
+  );
+};
+
+const Option = (props: OptionProps<ComponentFilterOption>) => {
+  const { label, isSelected, selectOption, data } = props;
+
+  const onClick = (e: MouseEvent) => {
+    selectOption(data);
+    e.stopPropagation();
+    e.preventDefault();
+  };
+
+  return (
+    <components.Option {...props}>
+      <TextButton className={styles.optionButton} onClick={onClick}>
+        <Checkbox label={label} checked={isSelected} />
+      </TextButton>
+    </components.Option>
+  );
+};
+
+const MenuList = (props: MenuListProps<ComponentFilterOption>) => {
+  const { children, getValue, hasValue, options, clearValue } = props;
+  const { t } = useTranslation();
+
+  const partiallyChecked = getValue().length !== options.length;
+
+  return (
+    <components.MenuList {...props}>
+      <div className={styles.option}>
+        <TextButton className={styles.optionButton} onClick={clearValue}>
+          <Checkbox
+            label={t('FILTER.SHOWING_ALL')}
+            checked={hasValue}
+            partiallyChecked={partiallyChecked}
+          />
+        </TextButton>
+      </div>
+      {children}
+    </components.MenuList>
+  );
+};
+
+export const FilterMultiSelectV2 = ({
+  subject,
+  options,
+  onChange,
+  isDisabled,
+  shouldReset,
+}: FilterMultiSelectProps) => {
+  const { t } = useTranslation();
+  const [value, setValue] = useState(() => options.filter((option) => option.default));
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const ref = useRef<HTMLDivElement | null>(null);
+  const hiddenCountRef = useRef<HTMLElement | null>(null);
+
+  useEffect(() => {
+    if (shouldReset) {
+      setValue(options.filter((option) => option.default));
+    }
+  }, [shouldReset]);
+
+  const sortedOptions = useMemo(() => {
+    const sorted = [...options].sort((a, b) => {
+      const isSelected = (option: ComponentFilterOption) => value.includes(option);
```

Comment:
Sorting doesn't work for me; was it working previously? The value array is always `[]` here for some reason.

Reply:
Sad! It did work at some point. Checking it

Reply:
So the problem was I was using `includes` but since the element I'm checking for is an object, it's not actually the same element. Should be fixed now!